General purpose delay provider

dbrgn opened the pull request:
Fixes #199 and improves docs.

MabezDev reviewed (approved):
I like this change, I've seen a few people stumble on this sort of thing. Thank you for improving the docs!

My nitpick would just be about the name, I would just call it `Delay`.

dbrgn commented:
@MabezDev I initially named it `Delay`, but then thought that it might not be ideal because it would collide with the `Delay` trait.

However, I just realized that the trait(s) are called `DelayMs` and `DelayUs`, so it shouldn't be an issue 🙂 I'll rename!

dbrgn commented:
Renamed, and I also fixed an embarrassing bug 😄

Additionally, I added the static methods on `Delay` and am calling these from the trait impls. This makes for a more convenient API and is consistent with the other delay providers.
